### ISSUE TITLE
mismatch on the attention formula in the paper and implementation

### ISSUE THREAD

**Issue by hwijeen** [facebookresearch/DME#6]
Hi, thanks for sharing the code! It really helped me understanding the paper.
I have a question on the way you calculate the attention score, especially on where to apply softmax on.
![2018-12-07 17 13 55](https://user-images.githubusercontent.com/29157715/49636257-6e98df80-fa45-11e8-9823-5a20c3364c96.png)

From this, it seems that the softmax function(g) takes as input "word vectors" FROM TIME STEP 1 TO S, meaning that alpha distribution over the SEQUENCE, not on NUMBER OF PRETRAINED VECTORS.

https://github.com/facebookresearch/DME/blob/97631c48e54390c81c60969ee4e74d0bf05e881b/dme/embedders.py#L183-L203

However, line 203 takes softmax over dim=2, which is NUMBER OF PRETRAINED VECTORS(self.n_emb).

I am little confused about the mismatch here. I think the formula on the paper should be revised. Am I missing something? Please help me!

**Comment by kahne** [comment]
Hi @hwijeen ,

For a fixed time step j, we have unnormalized attention weight for each embedding i. They are computed using the same function (the part inside function phi). The input of the function is the full sequence of word vectors from embedding i (for DME, only the vector at time step j is actually used; for CDME, all vectors are used). And then function phi (softmax/sigmoid) is applied to dimension i to normalize these weights.

Hope this answers your question.

**Comment by hwijeen** [comment]
> for DME, only the vector at time step j is actually used;

This answers my question. Thank you!
